libs/environment/src/hooks/use-environment.spec.tsx

```diff
@@ -0,0 +1,185 @@
+import type { ComponentProps } from 'react';
+import { renderHook } from '@testing-library/react-hooks';
+import { Networks } from '@vegaprotocol/react-helpers';
+import type { EnvironmentState } from './use-environment';
+import { useEnvironment, EnvironmentProvider } from './use-environment';
+import { ContractAddresses } from '../types';
+
+const MockWrapper = (props: ComponentProps<typeof EnvironmentProvider>) => {
+  return <EnvironmentProvider {...props} />;
+};
+
+const MOCK_HOST = 'https://vega.host/query';
+
+global.fetch = jest.fn();
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+const mockFetch = (url: RequestInfo) => {
+  if (url === mockEnvironmentState.VEGA_CONFIG_URL) {
+    return Promise.resolve({
+      ok: true,
+      json: () =>
+        Promise.resolve({
+          hosts: [MOCK_HOST],
+        }),
+    } as Response);
+  }
+  return Promise.resolve({ ok: true } as Response);
+};
+
+const mockEnvironmentState: EnvironmentState = {
+  configStatus: 'success',
+  VEGA_URL: 'https://vega.xyz',
+  VEGA_ENV: Networks.TESTNET,
+  VEGA_CONFIG_URL: 'https://vega.xyz/testnet-config.json',
+  VEGA_NETWORKS: {
+    TESTNET: 'https://testnet.url',
+    STAGNET: 'https://stagnet.url',
+    MAINNET: 'https://mainnet.url',
+  },
+  ETHEREUM_CHAIN_ID: 3,
+  ETHEREUM_PROVIDER_URL: 'https://ether.provider',
+  ETHERSCAN_URL: 'https://etherscan.url',
+  ADDRESSES: ContractAddresses[Networks.TESTNET],
+};
+
+beforeEach(() => {
+  // @ts-ignore typscript doesn't recognise the mock implementation
+  global.fetch.mockReset();
+  // @ts-ignore typscript doesn't recognise the mock implementation
+  global.fetch.mockImplementation(mockFetch);
+
+  window.localStorage.clear();
+
+  process.env['NX_VEGA_URL'] = mockEnvironmentState.VEGA_URL;
```

Comment:
question (blocking): I think we should probably tear down our mocked environment after these tests run

Reply:
We probably want to clear localstorage too

Reply:
done!